packs/tfe_fdo_nomad/README.md

```diff
@@ -0,0 +1,162 @@
+| `tfe_resource_memory`                        | no       | `"1024"`                                     | Memory in MB for TFE container.                                                                                                                                                                       |
+| `tfe_image`                                  | no       | `"hashicorp/terraform-enterprise:v202401-2"` | TFE image and tag to download and run.                                                                                                                                                                |
+| `tfe_image_registry_username`                | no       | `"terraform"`                                | The user name for the registry where the TFE image is hosted.                                                                                                                                         |
+| `tfe_image_server_address`                   | yes      | `"images.releases.hashicorp.com"`            | The server address of the registry where TFE image is hosted.                                                                                                                                         |
+| `tfe_run_pipeline_nomad_tls_config_insecure` | no       | `"false"`                                    | mTLS between Nomad and TFE when set to false.                                                                                                                                                         |
+| `tfe_agent_namespace`                        | no       | `"tfe-agents"`                               | Nomad namespace for TFE Agents to run.                                                                                                                                                                |
+| `tfe_agent_image`                            | no       | `"hashicorp/tfc-agent:latest"`               | TFE Agent image and tag to download and run.                                                                                                                                                          |
+| `tfe_vault_cluster_port`                     | no       | `"8201"`                                     | Vault cluster port which needs to exposed from the TFE container.                                                                                                                                     |
+| `tfe_vault_cluster_address`                  | no       | `"http://$${NOMAD_HOST_ADDR_vault}"`         | Cluster URL of the internal Vault server on this node (e.g., http://192.168.0.1:8201). Must be reachable across nodes.                                                                                |
```

Comment:
I'm not super familiar with TFE but the wording of this sticks out to me when I look at the Pack. The TFE job runs exposes a `network.port` named `"vault"` which is the `tfe_vault_cluster_port`, but there's also an address here which is also being referred to as belonging to the "cluster" but looks like it's actually supposed to be a Vault server not managed by Nomad? And why the wording "on this node"?

Reply:
https://developer.hashicorp.com/terraform/enterprise/flexible-deployments/install/configuration#tfe_vault_cluster_address

Each TFE container starts a vault server internally.
`tfe_vault_cluster_port` is to expose this port across all TFE instances.

All TFE containers access only the vault they brought up hence the word "on this node" which we can re-word to "on this container" but all vault instances need to be in sync and hence we need to expose the vault port.

Reply:
Ok, I understand. But in that case isn't this address _always_ going to be `"http://$${NOMAD_HOST_ADDR_vault}"`? Why make it configurable?

Reply:
In case you want to use a self-deployed vault cluster which is external to TFE.

Reply:
Ok, I'd probably change the description of this variable to make that clear then. Something like

> HTTP address of the Vault server used by TFE (ex. `"http://192.168.0.1:8201"`). By default, a Vault server is launched by this job. Must be reachable across nodes.

Dumb follow-up questions :grin: : 
* If a user did use a self-deployed Vault cluster, how do they tell the TFE container not to start its own?
* If a user did use a self-deployed Vault cluster, how does TFE get the TLS certs for it?

Reply:
Oh no, it's an awesome question. We are not giving users the option to do that as of now. If you want to do that you can use set the `TFE_VAULT_USE_EXTERNAL` and `TFE_VAULT_ADDRESS` environment variables with m/TLS certs mounted onto the containers.

We don't have support for that as of now in the pack and would take it as enhancements in future along with many other TFE env variables and flexibilities.